lib/player.rb

```diff
@@ -0,0 +1,95 @@
+require_relative 'scoring'
+require_relative 'tile_bag'
+
+module Scrabble
+
+  class Player
+
+    attr_reader :name, :plays, :total_score, :tiles
+
+    def initialize(name)
+      @name = name
+      @plays = []
+      @total_score = 0
+      @tiles = []
+    end
+
+    def string?(word)
+      return word.class == String
+    end #end of string?
+
+    def valid?(word)
+      count = 0
+      length = 0
+      for i in 0 ... word.length
+        if @tiles.include?(word[i].upcase.to_sym)
```

Comment:
I think one reason that this `valid?` method is not working properly is that we're not handling the case of duplicate letters in `word`.

In particular, if `@tiles` has the value `["B", "A", "N"]` and we call `valid?("BANANA")` it would return true. This is because we're check each letter in the word independently, when in fact for the last letter we need to check if it's included in `@tiles` _after_ all of the previous letters have been removed from it.

Here's a test which would expose this problem:

```ruby
it "should validate words with duplicate letters" do
  @person.draw_tiles(@tilebag)
  # Sanity check, for this test to work @person.tiles must have more than one distinct letter in it
  @person.tiles.uniq.length.must_be :>, 1

  test_word = @person.tiles.first * 7 # Create a word that is the first tile repeated 7 times
  @person.valid?(test_word).must_equal false
end
```